File: src/main/scala/uk/co/ridentbyte/view/dialog/BasicAuthInputDialog.scala
package uk.co.ridentbyte.view.dialog

import javafx.application.Platform
import javafx.geometry.HPos
import javafx.scene.control.{ButtonType, Dialog, Label, TextField}
import javafx.scene.layout.GridPane
import uk.co.ridentbyte.model.BasicAuth

class BasicAuthInputDialog extends Dialog[BasicAuth] {

  setTitle("Basic Auth")

  getDialogPane.getButtonTypes.addAll(ButtonType.CANCEL, ButtonType.OK)

  private val grid = new GridPane
  grid.setHgap(10)
  grid.setVgap(10)

  private val labelUsername = new Label("Username")
  GridPane.setHalignment(labelUsername, HPos.RIGHT)
  grid.add(labelUsername, 0, 0)

  private val textUsername = new TextField
  grid.add(textUsername, 1, 0)

  private val labelPassword = new Label("Password")
  GridPane.setHalignment(labelPassword, HPos.RIGHT)
  grid.add(labelPassword, 0, 1)

  private val textPassword = new TextField
  grid.add(textPassword, 1, 1)

  getDialogPane.setContent(grid)

  Platform.runLater(() => textUsername.requestFocus())

  setResultConverter(buttonType => {
    if (buttonType == ButtonType.OK) {
      BasicAuth(textUsername.getText, textPassword.getText)
    } else {
      null
    }
  })
}

File: src/main/java/uk/co/ridentbyte/view/dialog/BasicAuthInputDialog.java
package uk.co.ridentbyte.view.dialog;

import javafx.application.Platform;
import javafx.geometry.HPos;
import javafx.scene.control.ButtonType;
import javafx.scene.control.Dialog;
import javafx.scene.control.Label;
import javafx.scene.control.TextField;
import javafx.scene.layout.GridPane;
import uk.co.ridentbyte.model.BasicAuth;

public class BasicAuthInputDialog extends Dialog<BasicAuth> {

    public BasicAuthInputDialog() {
        this.setTitle("Basic Auth");
        this.getDialogPane().getButtonTypes().addAll(ButtonType.CANCEL, ButtonType.OK);

        GridPane grid = new GridPane();
        grid.setHgap(10);
        grid.setVgap(10);

        Label labelUsername = new Label("Username");
        GridPane.setHalignment(labelUsername, HPos.RIGHT);
        grid.add(labelUsername, 0, 0);

        TextField textUsername = new TextField();
        grid.add(textUsername, 1, 0);

        Label labelPassword = new Label("Password");
        GridPane.setHalignment(labelUsername, HPos.RIGHT);
        grid.add(labelPassword, 0, 1);

        TextField textPassword = new TextField();
        grid.add(textPassword, 1, 1);

        this.getDialogPane().setContent(grid);

        Platform.runLater(textUsername::requestFocus);

        this.setResultConverter((buttonType) -> {
            if (buttonType == ButtonType.OK) {
                return new BasicAuth(textUsername.getText(), textPassword.getText());
            } else {
                return null;
            }
        });
    }

}
